department-of-veterans-affairs/caseflow
# Add cancel field to cancel task modal

Resolves #14351

### Description
Adds a required reason field for the cancel task modal

### Acceptance Criteria
- [ ] Reason field shows in cancel task modal
- [ ] Cancel field is required
- [ ] Instructions provided in the reason field are saved to the task
- [ ] Instructions provided in the reason field are shown in case timeline

### Testing Plan
1. Sign in as BVASCASPER
2. Go to your queue and select an ama case
3. Select "cancel task" from the dropdown
4. Submit without instructions and notice error
5. Add instructions and press submit
6. Notice success alert and go back to case details
7. Ensure provided instructions are shown in case timeline

### UI Changes

BEFORE|AFTER|AFTER|AFTER|AFTER
---|---|---|---|---
![Screen Shot 2020-07-06 at 3 30 56 PM](https://user-images.githubusercontent.com/45575454/86632741-c854e900-bf9d-11ea-9a11-5de4889ef43a.png)|![Screen Shot 2020-07-06 at 3 09 58 PM](https://user-images.githubusercontent.com/45575454/86632373-4fee2800-bf9d-11ea-971c-42681c5d43ef.png)|![Screen Shot 2020-07-06 at 3 10 06 PM](https://user-images.githubusercontent.com/45575454/86632378-52e91880-bf9d-11ea-8851-f6c7cad3e213.png)|![Screen Shot 2020-07-06 at 3 20 21 PM](https://user-images.githubusercontent.com/45575454/86632387-54b2dc00-bf9d-11ea-838a-0d214d14bb2c.png)|<img width="608" alt="Screen Shot 2020-07-07 at 12 34 48 PM" src="https://user-images.githubusercontent.com/45575454/86815821-18d94e80-c051-11ea-846c-16d68639744a.png">
